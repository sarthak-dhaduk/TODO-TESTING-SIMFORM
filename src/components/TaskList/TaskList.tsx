import ToggledElement from "../InputAndButton/ToggledElement";
import TaskDisplay from "../DisplayTasks/TaskDisplay";

const tempObj = {
  completed: ["Buy new sweatshirt", "Begin promotional phase"],
  pending: [
    "Read an article",
    "Try not to sleep",
    "Watch 'Sherlock'",
    "Begin QA for the product",
    "Go for a walk",
  ],
};

function TaskList() {
  function newTask() {}

  return (
    <>
      <div className="h-[100%] overflow-auto ">
        <TaskDisplay list={tempObj.completed} type="completed" />
        <TaskDisplay list={tempObj.pending} type="pending" />
        {tempObj.pending.length <= 0 && tempObj.completed.length <= 0 && (
          <div className="flex flex-col items-center justify-center h-full">
            <div className="text-xl text-center">Nothing Left in the List!</div>
            <img
              src="/assets/hurray.jpg"
              alt="hurray"
              className="w-36 opacity-90"
            />
          </div>
        )}
      </div>
      <ToggledElement addNewTask={newTask} />
    </>
  );
}

export default TaskList;
