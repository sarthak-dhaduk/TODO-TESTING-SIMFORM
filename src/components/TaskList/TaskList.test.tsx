import "@testing-library/jest-dom/extend-expect";
import { render, screen } from "@testing-library/react";
import renderer from "react-test-renderer";
import TaskList from "./TaskList";

describe("TaskList component", () => {
  //testing the ui snapshot
  test("TaskList renders correctly", () => {
    const renderedComponent = renderer.create(<TaskList />).toJSON();
    expect(renderedComponent).toMatchSnapshot();
  });

  test("should render TaskDisplay components with correct props", () => {
    render(<TaskList />);
    const completedTasks = screen.getByText("Buy new sweatshirt");
    const pendingTasks = screen.getByText("Read an article");

    expect(completedTasks).toBeInTheDocument();
    expect(pendingTasks).toBeInTheDocument();
  });
});
