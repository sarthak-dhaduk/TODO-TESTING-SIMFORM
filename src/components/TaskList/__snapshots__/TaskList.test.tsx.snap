// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TaskList component TaskList renders correctly 1`] = `
Array [
  <div
    className="h-[100%] overflow-auto "
  >
    <ul>
      <li
        className="flex justify-between items-center mt-8"
      >
        <div
          className="break-words text-justify flex-1 mx-2 max-w-[80%] text-lg text-textComplete"
        >
          Buy new sweatshirt
        </div>
        <button
          className="w-9"
        >
          <img
            alt="completed"
            src="/assets/task-completed.png"
          />
        </button>
      </li>
      <li
        className="flex justify-between items-center mt-8"
      >
        <div
          className="break-words text-justify flex-1 mx-2 max-w-[80%] text-lg text-textComplete"
        >
          Begin promotional phase
        </div>
        <button
          className="w-9"
        >
          <img
            alt="completed"
            src="/assets/task-completed.png"
          />
        </button>
      </li>
    </ul>
    <ul>
      <li
        className="flex justify-between items-center mt-8"
      >
        <div
          className="break-words text-justify flex-1 mx-2 max-w-[80%] text-lg "
        >
          Read an article
        </div>
        <button
          className="w-9"
        >
          <img
            alt="completed"
            src="/assets/task-pending.png"
          />
        </button>
      </li>
      <li
        className="flex justify-between items-center mt-8"
      >
        <div
          className="break-words text-justify flex-1 mx-2 max-w-[80%] text-lg "
        >
          Try not to sleep
        </div>
        <button
          className="w-9"
        >
          <img
            alt="completed"
            src="/assets/task-pending.png"
          />
        </button>
      </li>
      <li
        className="flex justify-between items-center mt-8"
      >
        <div
          className="break-words text-justify flex-1 mx-2 max-w-[80%] text-lg "
        >
          Watch 'Sherlock'
        </div>
        <button
          className="w-9"
        >
          <img
            alt="completed"
            src="/assets/task-pending.png"
          />
        </button>
      </li>
      <li
        className="flex justify-between items-center mt-8"
      >
        <div
          className="break-words text-justify flex-1 mx-2 max-w-[80%] text-lg "
        >
          Begin QA for the product
        </div>
        <button
          className="w-9"
        >
          <img
            alt="completed"
            src="/assets/task-pending.png"
          />
        </button>
      </li>
      <li
        className="flex justify-between items-center mt-8"
      >
        <div
          className="break-words text-justify flex-1 mx-2 max-w-[80%] text-lg "
        >
          Go for a walk
        </div>
        <button
          className="w-9"
        >
          <img
            alt="completed"
            src="/assets/task-pending.png"
          />
        </button>
      </li>
    </ul>
  </div>,
  <button
    className="absolute bottom-[calc(11vh-30px)] left-[calc(50vw-38px)] rounded-full bg-lightGreen p-7 opacity-95"
    onClick={[Function]}
  >
    <img
      alt="plus"
      className="opacity-40 w-6"
      src="/assets/plus.png"
    />
  </button>,
]
`;
